### PR TITLE
Python upgrade

### LINKED ISSUE #461: Upgrade Invenio-accounts to v1.4.9 to enable account creation via CLI
Attempting to set up HEPData on a new dev machine and the initialise_db script fails to add the user to the DB as it isn't in the request context. v1.4.9 of `invenio-accounts` fixes this (see https://invenio-accounts.readthedocs.io/en/latest/changes.html) but will require upgrades of other Invenio modules including `invenio-base` (which upgrades Flask/Werkzeug to v2 so could require further changes).

### LINKED ISSUE #467: python: update to python > 3.6
python 3.6 reaches end of life in December 2021, so we should upgrade to a newer version. We currently use the [inspirehep python3.6 docker image](https://hub.docker.com/r/inspirehep/python/tags) as a base for the docker image used in production, so we will need to request a new version of that image, which might determine which version we use.

In the meantime we could try out various versions in CI.

### PR DESCRIPTION
**Pull request by alisonrclarke**
* Fixes #467 
 * Fixes #461 

Note that due to the celery upgrade, a change to the Kubernetes config will be needed before this is deployed: see   https://github.com/cern-sis/kubernetes/tree/hepdata-celery-v5 (and comments on #461).

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,26 +1,25 @@
 Babel==2.9.1
 beautifulsoup4==4.5.1
 bleach==3.3.0
-celery==4.4.6
+celery==5.1.2
```
Comment (GraemeWatt) [inline requirements.txt:4]: Could we upgrade to the latest version `celery==5.2.6`?  At least `5.2.2` is needed to fix the security vulnerability flagged in PR #473.

Comment (GraemeWatt) [reply requirements.txt:4]: Just checked, but `celery==5.2.2` requires `click>=8.0`, but `Flask==1.1.4` requires `click<8.0`.  So upgrading Celery further would require upgrading to Flask v2 (#348) and upgrades to other dependencies such as Invenio packages.  Let's leave it as `celery==5.1.2` and the Flask upgrade can be handled at a later stage.

**Review by GraemeWatt — APPROVED** [review]
Thanks for the quick work with the upgrade to Python 3.9.  I've successfully installed on my MacBook with Python 3.9 and made a few refinements to the installation docs.  I encountered a couple  of problems:
1. Trying to get MailCatcher working for the first time, I had to revert a change I made previously to get SMTP2GO to work.  The change is only needed when calling `starttls()` (i.e. if `SMTP_ENCRYPTION=True`).  This means mail is probably broken if the current `master` branch was to be deployed in production (where `SMTP_ENCRYPTION=False`).
2. Selenium v4 is needed when using Python 3.9 for the end-to-end tests to work properly.

Can you please check my changes before I merge?

**Comment by GraemeWatt** [comment]
> This means mail is probably broken if the current `master` branch was to be deployed in production (where `SMTP_ENCRYPTION=False`).

The current `master` branch is now deployed on QA and mail is working (with Python 3.6), so I can deploy in production before merging this PR.

**Review by alisonrclarke — COMMENTED** [review]
Your changes look good to me @GraemeWatt

**Pull request merged**